Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx

```diff
@@ -30,7 +30,7 @@ bool
 WriteSmallBmp(const std::string & filename)
 {
   const char    bmp_raw[] = { 66, 77, 30, 0, 0, 0, 0, 0, 0, 0,  26, 0, 0, 0,  12,
-                           0,  0,  0,  1, 0, 1, 0, 1, 0, 24, 0,  0, 0, -1, 0 };
+                           0,  0,  0,  1, 0, 1, 0, 1, 0, 24, 0,  0, 0, 255, 0 };
```

Comment:
maybe `static_cast<char>(-1)`?

Reply:
`char` may be signed or unsigned depending on the system. Perhaps changing the type to `unsigned char` in addition to this change  would be the most portable and explicit.

Reply:
Just test `unsigned char` on IBM power9 ppcle64. For original `-1`:

```
/rds/homes/y/yxz1148/ITK/Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx: In function ‘bool {anonymous}::WriteSmallBmp(const string&)’:
/rds/homes/y/yxz1148/ITK/Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx:40:20: error: invalid conversion from ‘const signed char*’ to ‘const char*’ [-fpermissive]
   bmp_output.write(bmp_raw, sizeof(bmp_raw));
                    ^~~~~~~
In file included from /rds/bear-apps/2019b/EL7-power9/software/GCCcore/8.3.0/include/c++/8.3.0/ostream:693,
                 from /rds/bear-apps/2019b/EL7-power9/software/GCCcore/8.3.0/include/c++/8.3.0/istream:39,
                 from /rds/bear-apps/2019b/EL7-power9/software/GCCcore/8.3.0/include/c++/8.3.0/sstream:38,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkMacro.h:53,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkLightObject.h:21,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkObject.h:31,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkRegion.h:31,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkImageRegion.h:31,
                 from /rds/homes/y/yxz1148/ITK/Modules/Core/Common/include/itkImage.h:21,
                 from /rds/homes/y/yxz1148/ITK/Modules/IO/BMP/test/itkBMPImageIOTestExtension.cxx:18:
/rds/bear-apps/2019b/EL7-power9/software/GCCcore/8.3.0/include/c++/8.3.0/bits/ostream.tcc:183:25: note:   initializing argument 1 of ‘std::basic_ostream<_CharT, _Traits>& std::basic_ostream<_CharT, _Traits>::write(const _CharT*, std::streamsize) [with _CharT = char; _Traits = std::char_traits<char>; std::streamsize = long int]’
     write(const _CharT* __s, streamsize __n)
           ~~~~~~~~~~~~~~^~~
make[2]: *** [Modules/IO/BMP/test/CMakeFiles/ITKIOBMPTestDriver.dir/itkBMPImageIOTestExtension.cxx.o] Error 1
make[1]: *** [Modules/IO/BMP/test/CMakeFiles/ITKIOBMPTestDriver.dir/all] Error 2
```

Reply:
Tried 3 ways to make it works.

```
  const  char bmp_raw[] = { 66, 77, 30, 0, 0, 0, 0, 0, 0, 0,  26, 0, 0, 0,  12,
                           0,  0,  0,  1, 0, 1, 0, 1, 0, 24, 0,  0, 0, static_cast<char>(-1) , 0 };
```

```
  const signed char bmp_raw[] = { 66, 77, 30, 0, 0, 0, 0, 0, 0, 0,  26, 0, 0, 0,  12,
                           0,  0,  0,  1, 0, 1, 0, 1, 0, 24, 0,  0, 0, -1 , 0 };

  bmp_output.write((char *)bmp_raw[0], sizeof(bmp_raw));
```

```
  const unsigned char bmp_raw[] = { 66, 77, 30, 0, 0, 0, 0, 0, 0, 0,  26, 0, 0, 0,  12,
                           0,  0,  0,  1, 0, 1, 0, 1, 0, 24, 0,  0, 0, 255 , 0 };

  bmp_output.write((char *)bmp_raw[0], sizeof(bmp_raw));
```

Reply:
Trying to submit PR: https://github.com/InsightSoftwareConsortium/ITK/pull/1965